pkg/controller/common/certificates/reconcile.go

```diff
@@ -32,8 +32,9 @@ type Reconciler struct {
 	K8sClient      k8s.Client
 	DynamicWatches watches.DynamicWatches
 
-	Object     metav1.Object       // owner for the TLS certificates (eg. Elasticsearch, Kibana)
-	TLSOptions commonv1.TLSOptions // TLS options of the object
+	Object         metav1.Object                     // owner for the TLS certificates (eg. Elasticsearch, Kibana)
+	TLSOptions     commonv1.TLSOptions               // TLS options of the object
+	ControllerSANs []commonv1.SubjectAlternativeName // SANs dynamically set by a controller, only used in the self signed cert
```

Comment:
Naming suggestion: I'd go for a more generic `ExtraHTTPSANs`. I feel like `ControllerSANs` is misleading: in practice this will contain a name based on the StatefulSet name, not anything related to the controller itself.